srvclient.go

```diff
@@ -34,6 +36,27 @@ func init() {
 	go dnsConfigLoop()
 }
 
+// SRVClient is a holder for methods related to SRV lookups. Parameters on it
+// can be modified before any methods are called, but not after. All methods are
+// thread-safe.
+type SRVClient struct {
+	// When initialized, SRVClient will cache the last successful SRV response
```

Comment:
Can there be a `InitCacheLast` function to simplify this? Then we can make it private. Maybe `EnableCacheLast` or something?